### PR TITLE
#160986722 fix jwt decode method

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
### What does this PR do?

Adds JWTAuthentication class to fix security issues with endpoints
Adds expiry time to JWT token
Fixes secret key configuration

#### Description of Task to be completed?

Currently, endpoints cannot be secured because tokens cannot be decoded

This PR decodes tokens so as to secure endpoints

#### How should this be manually tested?

Add an authentication class (authentication_classes = (JWTAuthentication,)) to an APIView to secure an endpoint
Add an authentication header to an endpoint that should be secured. Run the endpoint in postman.
Access to the endpoint should be restricted unless a token is provided

#### What are the relevant pivotal tracker stories?

- [#160986722](https://www.pivotaltracker.com/story/show/160986722)

### CODE REVIEW

**Comment by RonKbS** [comment]
Can you please specify how to secure the endpoint in your PR write up.
I've added 'authentication_classes = (JWTAuthentication,)' among the classes for the endpoint I wanted to secure for example.

**Review by RonKbS — APPROVED** [review]
I've pulled the branch and secured an endpoint and it LGTM.

**Comment by mendozabree** [comment]
@MuhanguziDavid Could you please change the title of your PR to the correct format 

```#[STORY_ID] [Story description]```

**Review by mendozabree — COMMENTED** [review]
There are some pep8 errors that you should look into.

**Comment by MuhanguziDavid** [comment]
I have looked into the pep8 issues and fixed them

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
 import jwt
-
```
Comment (lym) [inline authors/apps/authentication/models.py:2]: Please leave this empty line alone.

File: authors/apps/authentication/models.py [hunk 2]
```diff
@@ -102,12 +102,12 @@ def __str__(self):
 
     @property
     def get_full_name(self):
-      """
-      This method is required by Django for things like handling emails.
-      Typically, this would be the user's first and last name. Since we do
-      not store the user's real name, we return their username instead.
-      """
-      return self.username
+        """
```
Comment (lym) [inline authors/apps/authentication/models.py:105]: What's with this change? Why do you need to add extra spaces here?

**Comment by lym** [comment]
@MuhanguziDavid Please add tests for this so we don't decrease the coverage.

**Comment by lym** [comment]
@MuhanguziDavid Merging this but you have to create a bug-fix PR to create the tests.

File: authors/apps/authentication/models.py (hunk 2 above)
Comment (MuhanguziDavid) [reply authors/apps/authentication/models.py:105]: I added extra spaces to fix pep8 errors to do with improper indentation

File: authors/apps/authentication/models.py (hunk 1 above)
Comment (MuhanguziDavid) [reply authors/apps/authentication/models.py:2]: I was trying to group library imports together

**Pull request merged**